lib/pf/Switch/Meraki/AP_http_V2.pm

```diff
@@ -0,0 +1,358 @@
+# access technology supported
+sub description { 'Meraki cloud controller v2' }
+sub supportsWirelessMacAuth { return $TRUE; }
+sub supportsExternalPortal { return $TRUE; }
+sub supportsWirelessDot1x { return $TRUE; }
+sub supportsWebFormRegistration { return $FALSE }
+
+=head2 parseUrl
+
+This is called when we receive a http request from the device and return specific attributes:
+
+client mac address
+SSID
+client ip address
+redirect url
+grant url
+status code
+
+=cut
+
+sub parseUrl {
+    my($self, $req, $r) = @_;
+    my $logger = $self->logger;
+    
+    return ($$req->param('client_mac'),$$req->param('wlan'),$$req->param('client_ip'),$$req->param('redirect'),$$req->param('switch_url'),$$req->param('statusCode'));
+
+}
+
+=head2 deauthTechniques
+
+Return the reference to the deauth technique or the default deauth technique.
+
+=cut
+
+sub deauthTechniques {
+    my ($self, $method) = @_;
+    my $logger = $self->logger;
+    my $default = $SNMP::RADIUS;
+    my %tech = (
+        $SNMP::RADIUS => 'deauthenticateMacDefault',
+    );
+
+    if (!defined($method) || !defined($tech{$method})) {
+        $method = $default;
+    }
+    return $method,$tech{$method};
+}
+
+sub parseSwitchIdFromRequest {
+    my($class, $req) = @_;
+    return $$req->param('ap_mac');
+}
+
+=head2 returnRadiusAccessAccept
+
+Overloading L<pf::Switch>'s implementation to return specific attributes.
+
+=cut
+
+sub returnRadiusAccessAccept {
+    my ($self, $args) = @_;
+    my $logger = $self->logger;
+
+    my $radius_reply_ref = {};
+
+    $args->{'unfiltered'} = $TRUE;
+    my @super_reply = @{$self->SUPER::returnRadiusAccessAccept($args)};
+    my $status = shift @super_reply;
+    my %radius_reply = @super_reply;
+    $radius_reply_ref = \%radius_reply;
+
+    my @av_pairs = defined($radius_reply_ref->{'Cisco-AVPair'}) ? @{$radius_reply_ref->{'Cisco-AVPair'}} : ();
+    my $role = $self->getRoleByName($args->{'user_role'});
+    if(defined($role) && $role ne ""){
+        my $mac = $args->{'mac'};
+        my $node_info = $args->{'node_info'};
+        my $violation = pf::violation::violation_view_top($mac);
+        unless ($node_info->{'status'} eq $pf::node::STATUS_REGISTERED && !defined($violation)) {
+            my $session_id = generate_session_id(6);
+            my $chi = pf::CHI->new(namespace => 'httpd.portal');
+            $chi->set($session_id,{
+                client_mac => $mac,
+                wlan => $args->{'ssid'},
+                switch_id => $self->{_id},
+            });
+            pf::locationlog::locationlog_set_session($mac, $session_id);
+            my $redirect_url = $self->{'_portalURL'}."/cep$session_id";
+            $logger->info("Adding web authentication redirection to reply using role : $role and URL : $redirect_url.");
+            push @av_pairs, "url-redirect=".$redirect_url;
+
+        }
+
+    }
+
+    $radius_reply_ref->{'Cisco-AVPair'} = \@av_pairs;
+    $radius_reply_ref->{'Airespace-ACL-Name'} = $role;
```

Comment:
With that you force the radius attribute Airespace-ACL-Name, this is bad.
I think it should be better to handle it if RoleMap has been enabled.
You probably have to revert  your commit id dd76f84e5e9eeadb01dc86622df06ffd0019bcd4.